hasql/asyncpg.py

```diff
@@ -1,15 +1,46 @@
 import asyncio
-from typing import Sequence
+from typing import Dict, Optional, Sequence, Union
 
 import asyncpg  # type: ignore
+from packaging.version import parse as parse_version
 
-from hasql.base import BasePoolManager
+from hasql.base import (DEFAULT_ACQUIRE_TIMEOUT,
+                        DEFAULT_MASTER_AS_REPLICA_WEIGHT,
+                        DEFAULT_REFRESH_DELAY, DEFAULT_REFRESH_TIMEOUT,
+                        DEFAULT_STOPWATCH_WINDOW_SIZE, AbstractBalancerPolicy,
+                        BasePoolManager)
 from hasql.metrics import DriverMetrics
 from hasql.utils import Dsn
 
 
 class PoolManager(BasePoolManager):
     pools: Sequence[asyncpg.Pool]
+    cached_hosts: Dict[int, str]
+
+    def __init__(
```

Comment:
You can make cached_hosts as class var, and doesn't copy all constructor args and kwargs from BasePoolManager

Reply:
done

Reply:
Done, it has been fixed on the latest revision